bsms.mediawiki

```diff
@@ -63,39 +63,39 @@ The Signer is a participating member in the multisig. Its responsibilities inclu
+* The Coordinator calculates the <tt>MAC</tt> for the record. The <tt>MAC</tt> doubles as the <tt>IV</tt> for the encryption. 
+* The Coordinator encrypts the descriptor record with <tt>ENCRYPTION_KEY</tt> and <tt>IV</tt>. The Coordinator appends the MAC to the end of the encrypted text.
 * The Coordinator sends the encrypted descriptor record to all participating Signers.
 
 =====Signer=====
 
 * The Signer imports the descriptor record.
-* The Signer extracts the <tt>IV</tt> from the data, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> (derived from the open session) and <tt>IV</tt>.
-* The Signer verifies that the included <tt>IV</tt> is valid given the decrypted data.
-* The Signer calculates and verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
-* Abort the setup if there are any <tt>KEY</tt> duplicates.
-* The Signer checks whether one of the <tt>KEY</tt>s in the descriptor belongs to it, using path and fingerprint information included in the descriptor. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
-* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt>, plus important policy parameters, such as <tt>M</tt>, <tt>N</tt> and the position of its own XPUB in the policy script. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. The position is important for scripts where key order matters. Except for the position, all participating Signers should be able to display the same confirmation.
+* The Signer extracts the <tt>MAC</tt> from the data, sets <tt>IV</tt> to <tt>MAC</tt>, then decrypts the ciphertext using <tt>ENCRYPTION_KEY</tt> (derived from the open session) and <tt>IV</tt>.
+* The Signer verifies that the included <tt>MAC</tt> is valid given the plaintext.
+* The Signer verifies the descriptor’s <tt>CHECKSUM</tt>. Abort the setup if the <tt>CHECKSUM</tt> is incorrect.
+* The Signer checks that its <tt>KEY</tt> is included in the descriptor, using path and fingerprint information provided. The check must perform an exact match on the <tt>KEY</tt>s, and not using shortcuts such as matching fingerprints (which is trivial to spoof). Abort the setup if it doesn’t detect its own <tt>KEY</tt>.
+* For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt> and the policy parameters, such as <tt>M</tt>, <tt>N</tt> and the position of its own XPUB in the policy script. The total number of Signers, <tt>N</tt>, is important to prevent a <tt>KEY</tt> insertion attack. The position is important for scripts where key order matters. Except for the position, all participating Signers should be able to display the same confirmation.
```

Comment:
> I removed the "Abort the setup if there are any KEY duplicates" line because it occurred to me with Taproot and complex scripts, one can have the same XPUB in different taproot branches. HWWs should still check for duplicates for basic multisig, but we probably should leave this out of the spec.

For confirmation, the Signer must display to the user the <tt>CHECKSUM</tt> and the policy parameters, such as <tt>M</tt>, <tt>N</tt> and the position(s) of its own XPUB

Stating that (if applicable), all positions of the XPUB must be displayed will cover Taproot and key duplicate attacks for standard multisig

Confirming that the Signer must display all positions of the XPUB